ptrace.c

```diff
@@ -125,10 +125,11 @@ int ptrace_finish_attach(struct ptrace_child *child, pid_t pid) {
     memset(child, 0, sizeof *child);
     child->pid = pid;
 
-    kill(pid, SIGCONT);
     if (ptrace_wait(child) < 0)
         goto detach;
 
+    kill(pid, SIGCONT);
```

Comment:
Don't we need to also add a call to `waitpid()` after `kill(..., SIGCONT)`?

   waitpid(child->pid, &child->status, WCONTINUED);

If yes, maybe we could add an argument to `ptrace_wait()` to allow to pass the option `WCONTINUED` ?

Reply:
I think I understood your intention of handling signal-delivery-stop, but after pondering for a while, I realized maybe we don't need the kill(SIGCONT) at all?

1. `ptrace(ATTACH)`: the tracee will be put into t status (trace stop).
2. `waitpid()`: tracer (reredirecet) captures SIGSTOP so as to synchronize with tracee.
3. ~~kill(SIGCONT)~~: we don't need this? Correct me if I'm wrong, if we remove it, the tracee will be kept in t status, I don't see a problem for following steps.
4. `ptrace(PEEKUSER)`: inject dup syscalls and so on. 
5. `ptrace(SYSCALL)`: resume the tracee to execute injected dup syscall.
6. ...

Reply:
My local test went well by deleting `kill(SIGCONT)`, I updated the patch in case you want to try it too.